concourse/pipelines/modelbase.py

```diff
@@ -69,10 +70,28 @@ def custom_init(self, raw_dict: dict):
         if raw_dict.get('output_dir', None):
             name = raw_dict['output_dir']
             self.add_output(name + '_path', name + '_path')
+        # normalise step-arguments, if given. For convenience we allow both primitive integers and strings
+        # and lists containing entries of those two types.
+        args = raw_dict.get('arguments', None)
+        if args:
+            if isinstance(args, (str, int)):
+                args = [args]
+            for argument in args:
+                if not isinstance(argument, (str, int)):
+                    fail('Step arguments must be strings or integers, found: {t}'.format(t=type(argument)))
```

Comment:
do not "util.fail" outside of a "cli.py"-context! raise a ModelValidationError. This is a general rule.